features/situation-feedback/elastic/src/main/resources/OSGI-INF/blueprint/blueprint.xml

```diff
@@ -0,0 +1,102 @@
+        <cm:default-properties>
+            <!-- Elastic Connection Settings -->
+            <cm:property name="elasticUrl" value="http://localhost:9200" />
+            <cm:property name="globalElasticUser" value="" />
+            <cm:property name="globalElasticPassword" value="" />
+            <cm:property name="elasticIndexStrategy" value="monthly" />
+            <cm:property name="maxFlowDurationMs" value="120000" />
+            <cm:property name="defaultMaxTotalConnectionPerRoute" value="-1" />
+            <cm:property name="maxTotalConnection" value="-1" />
+            <cm:property name="nodeDiscovery" value="false" />
+            <cm:property name="nodeDiscoveryFrequency" value="0" />
+            <cm:property name="proxy" value=""/>
+            <cm:property name="retryCooldown" value="500" />
+            <cm:property name="connTimeout" value="5000" /> <!-- 5 second timeout for Elasticsearch operations -->
+            <cm:property name="readTimeout" value="30000" /> <!-- 30 second timeout for Elasticsearch socket reads -->
+            <cm:property name="retries" value="0" /> <!-- Disable retries by default -->
+            <cm:property name="nodeCache.enabled" value="true" /> <!-- Set to false to disable caching -->
+            <cm:property name="nodeCache.maximumSize" value="1000"/> <!-- Set value for unlimited size -->
+            <cm:property name="nodeCache.expireAfterWrite" value="300"/> <!-- in seconds. Set to 0 to never evict elements -->
+            <cm:property name="nodeCache.recordStats" value="true"/> <!-- Set to false to not expose cache statistics via jmx -->
+
+            <!-- Bulk Action Retry settings -->
+            <cm:property name="bulkRetryCount" value="5" /> <!-- Number of retries until a bulk operation is considered failed -->
+
+            <!-- Index settings -->
+            <!-- https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#index-modules-settings -->
+            <cm:property name="settings.index.number_of_shards" value="" />
+            <cm:property name="settings.index.number_of_replicas" value="" />
+            <cm:property name="settings.index.refresh_interval" value="" />
+            <cm:property name="settings.index.routing_partition_size" value="" />
+        </cm:default-properties>
+    </cm:property-placeholder>
+
+
+    <!-- Used to define credentials in a properties file and enforcing the bundle to reload when credentials change -->
+    <bean id="credentialsProvider" class="org.opennms.plugins.elasticsearch.rest.credentials.DefaultCredentialsProvider">
+        <argument value="elastic-credentials.xml" />
+    </bean>
+
+    <!-- Proxy for the JestClientFactory -->
+    <bean id="requestExecutorFactory" class="org.opennms.plugins.elasticsearch.rest.executors.DefaultRequestExecutor">
+        <argument value="${retryCooldown}"/>
+    </bean>
+    <bean id="clientFactory" class="org.opennms.plugins.elasticsearch.rest.RestClientFactory">
```

Comment:
On a general note, we may want to look at centralizing this configuration in order to avoid all of this boilerplate for each module that wants to persist something in Elasticsearch.